ocaml/xapi/import_raw_vdi.ml

```diff
@@ -111,20 +112,27 @@ let import vdi (req: Request.t) (s: Unix.file_descr) _ =
             | Some sr ->
               debug "Checking whether localhost can see SR: %s" (Ref.string_of sr);
               if (Importexport.check_sr_availability ~__context sr)
-              then localhost_handler rpc session_id vdi req s
-              else
+              then localhost_handler rpc session_id vdi req s  
+              else begin
+              try   
```

Comment:
I think it's best to wrap everything inside `Server_helpers.exec_with_new_task "VDI.import"` with this try/with block, between from line 104. This would make it even safer, because I think there are still some operations outside this try/with block that can fail. So we could write it like this:
```ocaml
   Server_helpers.exec_with_new_task "VDI.import"
     (fun __context ->
         try
           Helpers.call_api_functions ~__context
             (
                ...
             )
           )
         with
             ... error handling
     )
```
This way, we still have the context so we can mark the newly-created task as failed, but we protect as much code as possible with the try/with block.